manual/manual_compiler.rst

```diff
@@ -599,26 +647,65 @@ Optimisation level  Description
+We will now demonstrate the :py:meth:`default_compilation_pass` with the different levels of optimisation using the IBMQ Quito device. 
+
+As more intensive optimisations are applied by level 2 the pass may take a long to run for large circuits. In this case it may be preferable to apply the lighter optimisations of level 1.
+
+.. jupyter-input::
 
     from pytket import Circuit, OpType
-    from pytket.extensions.qiskit import AerBackend
-    circ = Circuit(3)
-    circ.CZ(0, 1)
+    from pytket.extensions.qiskit import IBMQBackend
+
+    circ = Circuit(3) # Define a circuit to be compiled to the backend
+    circ.CX(0, 1)
     circ.H(1)
     circ.Rx(0.42, 1)
     circ.S(1)
-    circ.add_gate(OpType.YYPhase, 0.96, [1, 2])
+    circ.CX(0, 2)
+    circ.CX(2, 1)
+    circ.Z(2)
+    circ.Y(1)
     circ.CX(0, 1)
+    circ.CX(2, 0)
     circ.measure_all()
-    b = AerBackend()
+
+    backend = IBMQBackend("ibmq_quito") # Initialise Backend
+
+    print("Total gate count before compilation =", circ.n_gates)
+    print("CX count before compilation =",  circ.n_gates_of_type(OpType.CX))
+
+     # Now apply the default_compilation_pass at different levels of optimisation.
+
     for ol in range(3):
-        test = circ.copy()
-        b.default_compilation_pass(ol).apply(test)
-        assert b.valid_circuit(test)
+        test_circ = circ.copy()
+        backend.default_compilation_pass(optimisation_level=ol).apply(test_circ)
+        assert backend.valid_circuit(test_circ)
         print("Optimisation level", ol)
-        print("Gates", test.n_gates)
-        print("CXs", test.n_gates_of_type(OpType.CX))
+        print("Gates", test_circ.n_gates)
+        print("CXs", test_circ.n_gates_of_type(OpType.CX))
+    
+.. jupyter-output::
+
+    Total gate count before compilation = 13
+    CX count before compilation = 5
+    Optimisation level 0
+    Gates 22
+    CXs 8
+    Optimisation level 1
+    Gates 12
+    CXs 5
+    Optimisation level 2
+    Gates 6
+    CXs 1
+
+**Explanation**
+
+We see that compiling the circuit to IBMQ Quito at optimisation level 0 actaully increases the gate count. This is because IBM Quito has connectivity constraints which require additional CX gates to be added to validate the circuit.
+The single qubit gates in our circuit also need to be decomposed into the IBM gatset.
+
+We see that compiling at optimisation level 1 manages to reduce the CX count to 5. Our connectivity constraints are satisfied without increasing the CX gate count. Single qubit gates are also combined to reduce the overall gate count further.
+
+Finally we see that the default pass for optimisation level 2 manages to reduce the overall gate count to just 6 with only one CX gate. This is because more intensive optimisations are applied at this level including squashing passes that enable optimal two and three qubit circuits to be synthesised. Applying these more powerful passes comes with a runtime overhead that may be noticebale for larger circuits.
```

Comment:
```suggestion
Finally we see that the default pass for optimisation level 2 manages to reduce the overall gate count to just 6 with only one CX gate. This is because more intensive optimisations are applied at this level including squashing passes that enable optimal two- and three-qubit circuits to be synthesised. Applying these more powerful passes comes with a runtime overhead that may be noticeable for larger circuits.
```

Reply:
done